### PR TITLE
[PodTargetInstaller] Only set project, dylib, and compatibility versi…

### PR DESCRIPTION
**Pull request by segiddins**
…ons to valid, three integer values.

Closes https://github.com/CocoaPods/CocoaPods/issues/3887.

Depends upon https://github.com/CocoaPods/Core/pull/253

### CODE REVIEW

File: spec/unit/installer/target_installer/pod_target_installer_spec.rb [hunk 1]
```diff
@@ -324,6 +324,23 @@ module Pod
...
+              settings['CURRENT_PROJECT_VERSION'].should == project_version
+              settings['DYLIB_COMPATIBILITY_VERSION'].should == compatibility_version
+            end
+          end
+
+          handles['1.alpha', '1.0.0', '1']
+          handles['0.1-alpha', '0.1.0', '0.1.0']
+          handles['1.2.3.4', '1.2.3', '1']
+          handles['0.2.3.4', '0.2.3', '0.2.3']
+          handles['1.alpha.2', '1.0.0', '1']
```
Comment (neonichu) [inline spec/unit/installer/target_installer/pod_target_installer_spec.rb]: Shouldn't this be "1.0.2"?

Comment (segiddins) [inline spec/unit/installer/target_installer/pod_target_installer_spec.rb]: No, since `1.alpha.2` is really equivalent to `1.0.0-alpha.2`

Comment (neonichu) [inline spec/unit/installer/target_installer/pod_target_installer_spec.rb]: Ah, right

**Comment by neonichu** [comment]
:+1:

**Comment by alloy** [comment]
:+1:

**Pull request merged**